include: Add i2c helper methods for async implementation

hernangtz opened the pull request:
Added 3 helper methods for async calls as they exist for the blocking implementation

Signed-off-by: Hernan I. Gutierrez-Vazquez <quic_hernang@quicinc.com>

hernangtz committed:
include: Add i2c helper methods for async implementation

Added 3 helper methods for async calls as they exist for
the blocking implementation

Signed-off-by: Hernan I. Gutierrez-Vazquez <quic_hernang@quicinc.com>

teburd commented:
This needs to be rebased, it looks like a merge commit was added and this is the wrong way to update this PR. Please rebase your i2c changes on to main and force push, there should be no merges/merge commits.

hernangtz commented:
> This needs to be rebased, it looks like a merge commit was added and this is the wrong way to update this PR. Please rebase your i2c changes on to main and force push, there should be no merges/merge commits.

yes. I used the button in the gui to rebase and it added an extra commit, I will fix it.

hernangtz commented:
@carlescufi  I just rebased and all test have passed, would you mind taking a look at this PR?
@teburd have approved also after I rebased and repushed.

hernangtz commented:
> @teburd please take another look

I had to rebase/repush as branch went behind again.

MaureenHelm commented on include/zephyr/drivers/i2c.h:
```diff
@@ -690,6 +690,103 @@ static inline int i2c_transfer_cb(const struct device *dev,
+ *
+ * @return a value from i2c_transfer_cb()
+ */
+static inline int i2c_transfer_dt_cb(const struct i2c_dt_spec *spec,
```
This should be `i2c_transfer_cb_dt`

hernangtz replied:
ok, will fix in few

MaureenHelm commented on include/zephyr/drivers/i2c.h:
```diff
@@ -690,6 +690,103 @@ static inline int i2c_transfer_cb(const struct device *dev,
+ *
+ * @return a value from i2c_write_read_cb()
+ */
+static inline int i2c_write_read_dt_cb(const struct i2c_dt_spec *spec, struct i2c_msg *msgs,
```
This should be `i2c_write_read_cb_dt`

teburd commented:
@MaureenHelm good catches
